fix(core): extend RAII locker lifetime until end of function

anthonybilinski opened the pull request:
Before it was being constructed then destroyed immediately, not
actually keeping the mutex locked.

- [x] Commits follow our [git commit guidelines](https://github.com/qTox/qTox/blob/master/CONTRIBUTING.md#git-commit-guidelines)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/6340)
<!-- Reviewable:end -->

bodwok reviewed (commented):
The mutex is now locked in scope according to RAII <img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/6340#-:-MbVQHrc0s0UthQIZqxE:bflka0k)* status: :shipit: complete! 1 of 1 LGTMs obtained



<!-- Sent from Reviewable.io -->

anthonybilinski committed:
fix(core): extend RAII locker lifetime until end of function

Before it was being constructed then destroyed immediately, not
actually keeping the mutex locked.
